sklearn/tests/test_pipeline.py

```diff
@@ -348,6 +350,18 @@ def test_pipeline_score_samples_pca_lof():
     assert_allclose(pipe.score_samples(X), lof.score_samples(pca.transform(X)))
 
 
+def test_score_samples_on_pipeline_without_score_samples():
+    # Test that a pipeline does not have score_samples method when the final
+    # step of the pipeline does not have score_samples defined.
+    pca = PCA(svd_solver='full', n_components='mle', whiten=True)
+    clf = LogisticRegression()
+    pipe = Pipeline([('pca', pca), ('clf', clf)])
+    assert_raises_regex(
```

Comment:
we're trying to avoid `assert_raises_regex` and use `with pytest.raises():` instead

Reply:
I have updated the code accordingly.